### PR TITLE
Create validation file & define validation function

### LINKED ISSUE #68: Create a reusable validation function and implement in signUp
## Description

We need to create a reusable function for malicious string validation to enhance security. This function will be used to prevent malicious JSON strings from being sent to the database. The validation function will use regular expressions to check for common malicious patterns like {}, [], and more.

Implement this function firstly for **usernames** in signUp.

## Requirements

- [x] Create a new file for validation.
- [x] Create the reusable validation function.
- [x] Call that function in the signup service file.

### PR DESCRIPTION
**Pull request by oddpharmacy**
# Description

I've created a function named **isSafeInput** to check for potential malicious strings, using the common 'bad' characters such as {}, (), ', ", and etc. You can find this function in the validation.ts file under utils. Then, I called that function in the _signup.service.ts_ file to check for valid usernames. An error will be thrown should such malicious characters be detected. Error handling improvements are necessary.

# Closes issue(s)

Resolves #68

# How to test
n/a

# Changes include

1. Create validation.ts file and isSafeInput function.
2. Called isSafeInput function in signup service.

# Checklist

- [x] I have tested this code

### CODE REVIEW

**Review by FefesG5 — CHANGES_REQUESTED** [review]
Hi Lisa looks good,

Wondering if you can add ' ; ' into the const malicious pattern. Seems many SQL injections have this symbol ' ; ' inside the injection string.  (semi colon)

File: src/service/signup.service.ts [hunk 1]
```diff
@@ -13,6 +14,14 @@ export default {
       email,
       uid,
     }: { username: string; email: string; uid: string } = userdata;
+
+    // Validate username using the isSafeInput function
+    if (!validation.isSafeInput(username)) {
+      throw new Error(
+        'Invalid input. Please check your input for malicious characters.',
```
Comment (Anissa3005) [inline src/service/signup.service.ts:21]: This is to explicit to send back to the user. "Please check your inputs" is enough. To explicit error messages (send to the user) can be dangerous.

**Review by Anissa3005 — CHANGES_REQUESTED** [review]
Great job. 
The only change (besides from Chai's request) is to make the error message less explicit.

**Review by FefesG5 — APPROVED** [review]
All good. Semicolon added.

**Review by Anissa3005 — APPROVED** [review]
LGTM 🚀

**Review by shonancanuck — APPROVED** [review]
Looks good to me

**Pull request merged**